src/transformers/models/idefics/modeling_idefics.py

```diff
@@ -1435,6 +1460,54 @@ def forward(
             attentions=outputs.attentions,
         )
 
+    def _prepare_encoder_decoder_kwargs_for_generation(self, inputs_tensor, model_kwargs, model_input_name):
+        pixel_values = model_kwargs.get("pixel_values", None)
+        image_encoder_embeddings = model_kwargs.get("image_encoder_embeddings", None)
+        model_kwargs["perceiver_embeddings"] = model_kwargs.get("perceiver_embeddings", None)
+
+        if pixel_values is not None:
+            batch_size, num_images = pixel_values.shape[:2]
+            pixel_values = pixel_values.contiguous().view(batch_size * num_images, *pixel_values.shape[2:])
+            image_encoder_embeddings = self.model.vision_model(pixel_values=pixel_values).last_hidden_state
+
+        elif image_encoder_embeddings is not None:
+            batch_size, num_images, image_seq_len, image_hidden_size = image_encoder_embeddings.size()
+            image_encoder_embeddings = image_encoder_embeddings.view(
+                batch_size * num_images, image_seq_len, image_hidden_size
+            )
+
+        if self.config.use_resampler:
+            if model_kwargs["perceiver_embeddings"] is None:
+                model_kwargs["perceiver_embeddings"] = self.model.perceiver_resampler(image_encoder_embeddings)
+
+            image_seq_len, image_hidden_size = model_kwargs["perceiver_embeddings"].size(1), model_kwargs[
+                "perceiver_embeddings"
+            ].size(2)
+            model_kwargs["perceiver_embeddings"] = model_kwargs["perceiver_embeddings"].view(
+                batch_size, num_images, image_seq_len, image_hidden_size
+            )
```

Comment:
do the resizes before assigning it into `model_kwargs["perceiver_embeddings"]`

Reply:
Sure I'll change that